core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/IntelliJResolver.scala

```diff
@@ -37,9 +43,44 @@ case class IntelliJPatternResolver(pattern: String) extends IntelliJResolver {
       "version" -> version.releaseOrBuild,
       "release" -> version.release.getOrElse("snapshot-release")
     )
-    val replaced = replacements.foldLeft(pattern) { case (path, (pattern, replacement)) =>
+    val replacedBeforeResolvingGlobs = replacements.foldLeft(pattern) { case (path, (pattern, replacement)) =>
       path.replace(s"[$pattern]", replacement)
     }
+    val replaced = replaceGlobsWithExistingDirectories(List(replacedBeforeResolvingGlobs)).head
     Dependency(replaced)
   }
+
+  // solution below assumes that each * character is used to mark one part of the path (one atomic directory),
+  // for example: "file:///home/.cache/ides/com.jetbrains.intellij.idea/ideaIC/[revision]/*/ideaIC-[revision]/"
+  @tailrec
+  private def replaceGlobsWithExistingDirectories(paths: List[String]): List[String] =
+    if (paths.exists(pathMightBeValidResource))
+      paths.filter(pathMightBeValidResource)
+    else
+      replaceGlobsWithExistingDirectories(paths.flatMap(replaceFirstFoundWildcardWithDirectories))
+
+  private def pathMightBeValidResource(path: String): Boolean =
+    path.indexOf('*') == -1 &&
+      (path.endsWith(".zip") ||
+        path.endsWith(".dmg") ||
+        Paths.get(path.replace("file:", "")).isDirectory)
+
+  private def replaceFirstFoundWildcardWithDirectories(path: String): List[String] = {
+    def removeLastFileSeparator(path: String): String = if (path.endsWith("/")) path.init else path
+
+    val pathUntilWildcard = Paths.get(path.substring(0, path.indexOf('*')).replace("file:", ""))
+    val stringPathAfterWildcard = path.substring(path.indexOf('*') + 1)
```

Comment:
can you use split for that? It allows to specify limit of splits.

Reply:
Ok, but then I would have to use three `val`s instead of two:
```
val Array(stringPathUntilWildcard, stringPathAfterWildcard) = path.split('*')
val pathUntilWildcard = Paths.get(stringPathUntilWildcard.replace("file:", ""))
```
If you think that would be better than current solution, I can do that

Reply:
Yes, let's do this. It is same number of lines, but 2 substring calls with some index offset are more complex for reading and understanding than a simple split.

Reply:
There is a problem connected to replacing `String#indexOf` with `String#split`
The `String#split(String regex, int limit)` method treats the String parameter `"*"` as a regular expression, which leads to invalid results. 

I would suggest to switch back to previous logic:
```
val pathUntilWildcard = Paths.get(path.substring(0, path.indexOf('*')).replace("file:", ""))
val stringPathAfterWildcard = path.substring(path.indexOf('*') + 1)
```
to be sure that `*` wildcard will not be used as a regular expression.

Reply:
To be sure, just use regex as expected r"\*" or "[*]"

Reply:
Ok, but the thing is - I don't want java String's method to treat the `*` as a regular expression. And `String#split` by default treats the String parameter as a regular expression. That's why I want to stick to current logic, without using `split`. So that `*` will never be treated as a regex.